pkg/disk/partition_table.go

```diff
@@ -678,6 +689,59 @@ func (pt *PartitionTable) ensureLVM() error {
 	return nil
 }
 
+// ensureBtrfs will ensure that the root partition is on a btrfs subvolume, i.e. if
+// it currently is not, it will wrap it in one
+func (pt *PartitionTable) ensureBtrfs() error {
+
+	rootPath := entityPath(pt, "/")
+	if rootPath == nil {
+		return fmt.Errorf("no root mountpoint for a partition table: %#v", pt)
+	}
+
+	// we need a /boot partition to boot btrfs, ensure one exists
+	bootPath := entityPath(pt, "/boot")
+	if bootPath == nil {
+		_, err := pt.CreateMountpoint("/boot", 512*common.MiB)
+		if err != nil {
+			return fmt.Errorf("cannot create /boot partition when ensuring btrfs: %w", err)
```

Comment:
The phrasing of this error implies that creating a boot partition during `ensureBtrfs()` isn't possible at all (`cannot create ...`) rather than saying that something went wrong (e.g. `could not create ...` or `failed to create ...`).